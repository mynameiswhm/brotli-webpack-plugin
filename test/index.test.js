var BrotliPlugin = require('../index.js');
var webpack = require('webpack');
var rmRf = require('rimraf');

var OUTPUT_DIR = __dirname + '/tmp/';

function createCompiler(options) {
    var defaultOptions = {
        bail: true,
        cache: false,
        entry: __dirname + '/fixtures/entry.js',
        output: {
            path: OUTPUT_DIR,
            filename: '[name].js',
            chunkFilename: '[id].[name].js'
        },
        plugins: [
            new BrotliPlugin()
        ]
    };

    if (!Array.isArray(options)) {
        options = Object.assign({}, defaultOptions, options);
    }

    return webpack(options);
}

function compile(compiler) {
    return new Promise(function (resolve, reject) {
        compiler.run(function (err, stats) {
            if (err) return reject(err);
            resolve(stats);
        });
    });
}

describe('when applied with default settings', function () {
    beforeEach(function (done) {
        rmRf(OUTPUT_DIR, done)
    });

    afterEach(function (done) {
        rmRf(OUTPUT_DIR, done)
    });

    it('matches snapshot', function () {
        var compiler = createCompiler();

        new BrotliPlugin().apply(compiler);

        return compile(compiler).then(function (stats) {
            expect(stats.compilation.assets['main.js']);
            expect(stats.compilation.assets['main.js.br']);

            for (var file in stats.compilation.assets) {
                if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
                    expect(stats.compilation.assets[file].source()).toMatchSnapshot('default: ' + file);
                }
            }
        });
    });
});
