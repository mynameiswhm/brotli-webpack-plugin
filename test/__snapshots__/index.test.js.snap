// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with default settings matches snapshot: default: main.js 1`] = `"!function(e){var n={};function t(r){if(n[r])return n[r].exports;var o=n[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=e,t.c=n,t.d=function(e,n,r){t.o(e,n)||Object.defineProperty(e,n,{configurable:!1,enumerable:!0,get:r})},t.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},t.n=function(e){var n=e&&e.__esModule?function(){return e.default}:function(){return e};return t.d(n,\\"a\\",n),n},t.o=function(e,n){return Object.prototype.hasOwnProperty.call(e,n)},t.p=\\"\\",t(t.s=0)}([function(e,n){e.exports=function(){console.log(7)}}]);"`;

exports[`when applied with default settings matches snapshot: default: main.js.br 1`] = `
Object {
  "data": Array [
    27,
    68,
    2,
    0,
    228,
    169,
    105,
    211,
    154,
    115,
    154,
    145,
    16,
    186,
    76,
    29,
    101,
    24,
    163,
    189,
    180,
    127,
    146,
    142,
    175,
    240,
    213,
    97,
    224,
    156,
    32,
    163,
    217,
    208,
    166,
    28,
    21,
    84,
    150,
    148,
    22,
    105,
    251,
    27,
    143,
    39,
    105,
    182,
    95,
    117,
    83,
    161,
    8,
    6,
    150,
    181,
    31,
    45,
    184,
    231,
    234,
    66,
    65,
    172,
    69,
    17,
    35,
    83,
    206,
    62,
    135,
    61,
    182,
    216,
    64,
    190,
    157,
    163,
    174,
    155,
    45,
    72,
    137,
    116,
    158,
    44,
    101,
    230,
    48,
    37,
    113,
    248,
    17,
    231,
    219,
    172,
    193,
    231,
    172,
    24,
    97,
    66,
    60,
    203,
    238,
    115,
    32,
    216,
    17,
    100,
    15,
    70,
    76,
    248,
    43,
    76,
    184,
    150,
    13,
    218,
    205,
    147,
    139,
    50,
    158,
    119,
    163,
    23,
    11,
    206,
    160,
    15,
    214,
    205,
    39,
    151,
    219,
    68,
    197,
    6,
    178,
    81,
    90,
    122,
    240,
    126,
    43,
    196,
    145,
    37,
    196,
    251,
    80,
    104,
    234,
    58,
    127,
    233,
    212,
    61,
    61,
    165,
    190,
    72,
    191,
    94,
    18,
    249,
    165,
    196,
    139,
    216,
    172,
    113,
    112,
    52,
    106,
    88,
    142,
    65,
    172,
    113,
    216,
    239,
    165,
    253,
    169,
    34,
    84,
    204,
    146,
    122,
    179,
    162,
    100,
    1,
    180,
    45,
    232,
    157,
    12,
    6,
    66,
    184,
    229,
    242,
    182,
    227,
    67,
    23,
    178,
    226,
    144,
    207,
    18,
    72,
    17,
    191,
    24,
    125,
    70,
    197,
    112,
    12,
    168,
    128,
    250,
    72,
    120,
    50,
    81,
    97,
    120,
    245,
    228,
    169,
    40,
    210,
    245,
    216,
    174,
    126,
    116,
    31,
    152,
    17,
    8,
    8,
    225,
    249,
    75,
    197,
    33,
    160,
    69,
    163,
    150,
    75,
    240,
    184,
    102,
    21,
    196,
    142,
    214,
    121,
    195,
    175,
    114,
    142,
    158,
    233,
    18,
    199,
    56,
    223,
    183,
    48,
    7,
  ],
  "type": "Buffer",
}
`;
